### PR TITLE
#159951992 Upgrade Authors Haven Django app from Django 1.10 to Django 2.1

### PR DESCRIPTION
**Pull request by reiosantos**
#### What does this PR do?
- It upgrades the codebase from django ==1.10== to django 2.1

#### Description of Task to be completed?
- review all files to ensure that they are updated to match the new version

#### How should this be manually tested?
- Install django version 2.1
- After cloning the repo, cd into the directory and RUN ```python python manage.py runserver ```
- It should have resolved all errors and running the server

#### What are the relevant pivotal tracker stories?
[#159951992](https://www.pivotaltracker.com/story/show/159951992)

### CODE REVIEW

**Comment by inno-asiimwe** [comment]
@reiosantos we agreed to follow a one commit convention, no?

**Review by inno-asiimwe — COMMENTED** [review]
@reiosantos could you use the new Dango 2 url formats?

**Comment by reiosantos** [comment]
In my latest commit, I have used the new Django 2 URLs
and have also squashed the commits into one

File: authors/apps/authentication/urls.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
     LoginAPIView, RegistrationAPIView, UserRetrieveUpdateAPIView
 )
 
 urlpatterns = [
-    url(r'^user/?$', UserRetrieveUpdateAPIView.as_view()),
-    url(r'^users/?$', RegistrationAPIView.as_view()),
-    url(r'^users/login/?$', LoginAPIView.as_view()),
+    re_path('user/?$', UserRetrieveUpdateAPIView.as_view()),
+    re_path('users/?$', RegistrationAPIView.as_view()),
+    re_path('users/login/?$', LoginAPIView.as_view()),
```
Comment (inno-asiimwe) [inline authors/apps/authentication/urls.py]: Why are we using `re_path` instead of `path`?

Comment (reiosantos) [reply authors/apps/authentication/urls.py]: **re_path()** allows me to write regex expressions in the URL. while **path()** does not allow that. and the URLs in authentication are regex

Comment (inno-asiimwe) [reply authors/apps/authentication/urls.py]: Uniformity is key @reiosantos , please refactor the urls so that we use path throughout the application.

**Comment by huxaiphaer** [comment]
@reiosantos , check your spellings `It should have resoplved all errors and runing the server` in your PR,

**Pull request merged**